topi/python/topi/vision/yolo2/region.py

```diff
@@ -0,0 +1,64 @@
+"""
+REGION Operator
+====================
+**Author**: `Siju Samuel <https://github.com/siju-samuel/>`_
+Region operator, used in darknet.
+"""
+from __future__ import absolute_import as _abs
+import tvm
+import topi
+
+@tvm.target.generic_func
+def region(data, num, classes, coords, background, _):
+    """Region forward operators.
+
+    Parameters
+    ----------
+    Input : tvm.Tensor
+        4-D with shape [batch, in_channel, in_height, in_width]
+
+    stride : int
+        Stride value for reorganization
+
+    Returns
+    -------
+    Output : tvm.Tensor
+        4-D with shape [batch, out_channel, out_height, out_width]
+    """
+
+    batch, c, _, _ = topi.util.get_const_tuple(data.shape)
+
+    split_indices = c
+    split_res = topi.transform.split(data, split_indices, 1)
+
+    def channel_select_activation(range_val, coords):
+        # apply activiation first 2 layers and for last coords
+        if range_val == 2:
+            return coords
+        return range_val
+
+    for num_it in range(num):
```

Comment:
the logic here is a bit confusing. If we reshape the data to [b, num, (classes + coords + 1), h, w], then we can get rid of iterating over num. And then we can split along dim=2 and apply sigmoid, softmax, and so on.

Reply:
Done